stdlib/public/core/ArrayCast.swift

```diff
@@ -152,7 +156,7 @@ internal func _arrayConditionalBridgeElements<SourceElement, TargetElement>(
   let buf = _ContiguousArrayBuffer<TargetElement>(
     uninitializedCount: source.count, minimumCapacity: 0)
   
-  var p = buf.firstElementAddress
+  var p = buf.firstElementAddress!
```

Comment:
The rule is "we are going to load unconditionally" and "startIndex == 0".  It is the latter part that makes the investigation tricky.